### ISSUE TITLE
Router config of static files directory does not work properly

### ISSUE THREAD

**Issue by pavsyk** [go-aah/aah#141]
If you want to place static files into other directory then "static" (see the excerpt from the routes.conf):
```    static {
      public_assets {
        path = "/assets"
        dir = "assets"
      }
      favicon {
        path = "/favicon.png"
        file = "img/favicon.png"
      }
```
Then the URL `http://localhost:8080/favicon.png` returns 404, because the framework is still trying to deliver the `static/img/favicon.png` file instead of `assets/img/favicon.png`. It seems that the "static" prefix is hardcoded here, but it should take the `static/public_assets/dir` config value instead.

**Comment by jeevatkm** [comment]
@pavsyk Thanks for reporting an issue. I will have a analyze and get back to you.

**Comment by jeevatkm** [comment]
@pavsyk Thanks, I have analyzed it. Sorry for late response, caught up with some work.

Yes, you're right. Can you use default directory named `static` and map path to `/assets`?

By design router and static file delivery supports many directories and files from relative to absolute paths. I need to think about flexible configuration for your described scenario.

For e.g.: One of my thought
```
public_assets {
  path = "/assets"
  dir = "assets"
}

public_downloads {
  path = "/file/downloads"
  dir = "/home/aah/downloads"
  list = true
}

favicon {
  path = "/favicon.png"
  file = "img/favicon.png"

  # by using references if multiple directories defined 
  # otherwise use the one defined automatically.
  base_dir = public_assets.dir   
}
```

**Comment by pavsyk** [comment]
@jeevatkm Thanks. Yes, it is what I did (i.e. mapped the `static` dir to `/assets` path) as a workaround.

**Comment by jeevatkm** [comment]
@pavsyk Thank you for your patience. I have implemented this enhancement, currently available on edge version. You can give it try.

To be released as part `v0.10` 😄 

_PS: BTW aah docker images are available https://hub.docker.com/r/aahframework/aah_
